Add manual section for requirements (#861)
virtualsatellite/VirtualSatellite4-Core: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc

You finalized the tutorial for the extended product structures. 

==== Requirements concept

> To support model based requirements management and to verify requirements by using the system model, Virtual Satellite provides a concept to model requirements. Requirements can be either imported from other tools or created directly within Virtual Satellite.

Author: There is now semantic-less tracing of requirements to the system model, only tracing as aprt of verification. For that we have a section in the chapter.

===== First Steps

Before you can start: 

* Open Virtual Satellite 

* Create a new Virtual Satellite Project

* Activate your required concepts, such as PS and requirements 

For more information check <<Modeling with Virtual Satellite>>.

===== Configure Requirements Modeling

:step: 0
[%header,cols=">8,62a,40a"] 
|===

|Step
|Task
|Image

|{counter:step}.
|Create a RequirementsConfigurationCollection element directly in the Virtual Satellite Repository. 

* This collection is a place where all requirements configuration elements should be stored

TIP: User rights can be customized to prevent other users from editing this collection 

> Add a RequirementsConfiguration under the collection

Reviewer: ALready fixed it

|image:chapterModeling/subsectionRequirements/AddRequirementType.png[Add requirement type]

|{counter:step}.
|Specify a name for the new requirement type and add attributes in the dialog box.
|image:chapterModeling/subsectionRequirements/SpecifyRequirementTypeNameAndAttributes.png[Specify requirement type name and attributes]

|===

===== Modeling of requirements

:step: 0
[%header,cols=">8,62a,40a"] 
|===

|Step
|Task
|Image

|{counter:step}.
|Go to the model location where you want to add requirements. 

NOTE: Requirements can also be split to several elements within a model. 

At the desired location create a _RequirementSpecification_.
|image:chapterModeling/subsectionRequirements/CreateRequirementSpec.png[Create requirement specification]

|{counter:step}.
|Double-click the specification element to open its editor. 

Add requirements by using the button `Add Requirement` in the requirements section.
|image:chapterModeling/subsectionRequirements/AddRequirement.png[Add requirements]

|{counter:step}.
|When a new requirement is added, the editor asks of which type it needs to be: 

* You can now choose between the types you create in the configuration element of your project. 

> The left part of the dialog allows to filter by the container element of the configuration, the right side show all applicable types in the currently selected tree element.

Reviewer 2: ```suggestion
The left part of the dialog allows to filter by the container element of the configuration, the right side shows all applicable types in the currently selected tree element.

```

|image:chapterModeling/subsectionRequirements/ChooseRequirementType.png[Choose requirement type]

|{counter:step}.
|After a requirement of the selected type is added, its values can be added.
|image:chapterModeling/subsectionRequirements/AddValues.png[Add values]

|{counter:step}.
|It is also possible to create hierarchies of requirements.
To do that, create a requirement group by clicking the button `Add RequirementGroup`.
|image:chapterModeling/subsectionRequirements/CreateRequirementGroup.png[Create requirement group]

|{counter:step}.
|To create requirements within a group click on the button `Drill-Down` to go to the group.
|image:chapterModeling/subsectionRequirements/DrillDownReqGroup.png[Drill-Down to requirement group]

|===

===== Import / export of requirements 

> To integrate requirements management into the model-based systems engineering process, Virtual Satellite supports import of requirements from external tools.

Reviewer 2: ```suggestion
To integrate requirements management into the model-based systems engineering process, Virtual Satellite supports importing requirements from external tools.

```

System model-based verification methods are provided via the different domain concepts.
For more information how concepts are developed check out our https://github.com/virtualsatellite/VirtualSatellite4-Core/releases/download/Release_4.12.1/VirSat_Core_Developer_Manual.pdf[developer manual]. 

====== Add verification methods to requirements

:step: 0
[%header,cols=">8,62a,40a"] 
|===

|Step
|Task
|Image

|{counter:step}.
|To add verification a method to a requirement open it in the VirSat Editor.
Easiest way to do so is to press the `Drill-Down` button in the _RequirementSpecification_ or by double-clicking the element in the VirSat Navigator.
|image:chapterModeling/subsectionRequirements/OpenRequirementEditor.png[Open requirement in editor]

|{counter:step}.
|Add verification methods in the dedicated editor sections of the requirement.
One requirement can also have multiple verification methods. 

NOTE: Basic verification types are created in the _VerificationConfiguration_.

NOTE: System model-based verification methods can be added via dedicated editor sections.

> Depending on the used concepts in the project there might be different of these model-based verification methods. 

Reviewer 2: ```suggestion
Depending on the used concepts in the project, there might be different model-based verification methods. 

```
